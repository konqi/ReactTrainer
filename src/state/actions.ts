export enum ActionType {
  ADD_TRAINEE
}
