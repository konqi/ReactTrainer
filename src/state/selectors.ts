import {ApplicationState} from '.'

export const selectTraineeById = (state: ApplicationState, traineeId: string) =>
  state.trainees[traineeId]
