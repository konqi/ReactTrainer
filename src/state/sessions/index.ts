export {
  createAddSessionAction as createAddTrainingAction,
} from './sessionActions'
export {initialSessionState} from './sessionReducer'
