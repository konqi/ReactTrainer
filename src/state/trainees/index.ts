export {reduceTrainees, initialTraineeState} from './traineeReducer'
export {
  TraineeActions,
  createIngestTraineeAction as createAddTraineeAction,
} from './traineeActions'
