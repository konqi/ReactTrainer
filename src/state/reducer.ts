import { ActionType } from "./actions";

export interface Action {
  type: ActionType;
  payload: Trainee;
}

export interface Trainee {
  name: string;
  price: number;
}

export interface ApplicationState {
  trainees: Trainee[];
}

export const initialApplicationState: ApplicationState = {
  trainees: []
};

export const reducer = (
  state: ApplicationState,
  action: Action
): ApplicationState => {
  switch (action.type) {
    case ActionType.ADD_TRAINEE:
      const newTrainees = [...state.trainees, action.payload];
      return { ...state, trainees: newTrainees };
    default:
      throw new Error();
  }
};
