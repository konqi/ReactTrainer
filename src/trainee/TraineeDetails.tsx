import {
  Container,
  createStyles,
  Grid,
  makeStyles,
  Paper,
  Theme,
} from '@material-ui/core'
import {grey} from '@material-ui/core/colors'
import React, {useMemo, useState} from 'react'
import {useDispatch, useSelector} from 'react-redux'
import {ApplicationState} from '../state'
import {createAddTrainingIntend} from '../state/intends/UserIntend'
import {findFirstBeforeAndAfterDate, Session} from '../types/Session'
import {Trainee} from '../types/Trainee'
import {NewSession} from './NewSession'
import {TrainingEntry} from './TrainingEntry'
import {TraineeBalance, TraineeHistory} from './'

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    root: {
      padding: theme.spacing(3, 2),
    },
    calendar: {
      marginBottom: theme.spacing(2),
    },
    current: {
      backgroundColor: grey[200],
      borderRadius: `${theme.shape.borderRadius}px ${
        theme.shape.borderRadius
      }px 0px 0px`,
    },
    form: {
      backgroundColor: grey[200],
      paddingTop: theme.spacing(2),
      paddingBottom: theme.spacing(2),
      marginBottom: theme.spacing(2),
      borderRadius: theme.shape.borderRadius,
    },
  })
)

interface ExternalProps {
  traineeId: string
}

export const TraineeDetails: React.FC<ExternalProps> = ({traineeId}) => {
  const trainee = useSelector<ApplicationState, Trainee>(state => {
    return state.trainees[traineeId]
  })
  const {sessions} = useSelector<
    ApplicationState,
    {sessions: Session[] | undefined}
  >(state => {
    const sessions =
      state.trainees[traineeId] &&
      state.trainees[traineeId].sessionsRef &&
      state.trainees[traineeId].sessionsRef!.map(
        (sessionRef: string) => state.sessions[sessionRef]
      )
    // TODO add next & previous session
    return {sessions, nextSession: undefined, previousSession: undefined}
  })
  const price = (trainee && trainee.price) || 0
  const dispatch = useDispatch()
  const [datetime, setDatetime] = useState(new Date())
  const [description, setDescription] = useState('')
  const [payedAmount, setPayedAmount] = useState(price)
  const classes = useStyles()

  const {previousSession, upcomingSession} = useMemo(
    () => findFirstBeforeAndAfterDate(new Date(), sessions),
    [sessions]
  )

  const addNewTraining = () => {
    if (trainee) {
      dispatch(
        createAddTrainingIntend(
          {
            datetime,
            payedAmount,
            description,
          },
          trainee.id
        )
      )
    }
  }

  const handleChange = (
    propertyName: string,
    value: number | string | Date
  ) => {
    switch (propertyName) {
      case 'date':
      case 'time':
        return setDatetime(value as Date)
      case 'payedAmount':
        return setPayedAmount(value as number)
      case 'description':
        return setDescription(value as string)
    }
  }

  return (
    <Container className={classes.root}>
      <Paper>
        <Container>
          <Grid container spacing={3}>
            <Grid item xs={4} className={classes.calendar}>
              letztes mal
              {previousSession && (
                <TrainingEntry date={previousSession.datetime} />
              )}
            </Grid>
            <Grid item xs={4} className={classes.current}>
              aktuell
              <TrainingEntry date={datetime} />
            </Grid>
            <Grid item xs={4} className={classes.calendar}>
              n√§chstes mal
              {upcomingSession && (
                <TrainingEntry date={upcomingSession.datetime} />
              )}
            </Grid>
          </Grid>
          <Grid container>
            <Grid item xs={12} className={classes.form}>
              <NewSession
                datetime={datetime}
                payedAmount={payedAmount}
                description={description}
                onChange={handleChange}
                onSubmit={addNewTraining}
              />
            </Grid>
          </Grid>
        </Container>
      </Paper>

      <TraineeBalance traineeId={traineeId} />
      <TraineeHistory traineeId={traineeId} />
    </Container>
  )
}

// receive payment

// show balance (underpayed / overpayed), estimate? remaining session count

// option to cancel / delete entry
