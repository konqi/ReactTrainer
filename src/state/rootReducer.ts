import {ApplicationState} from '.'
import {FSA} from '../types/FSA'
import {reduceTrainees} from './trainees'

export const rootReducer = (
  state: ApplicationState,
  action: FSA<any, any>
): ApplicationState => ({
  trainees: reduceTrainees(state.trainees, action),
})
