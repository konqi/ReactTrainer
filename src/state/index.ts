import {TraineeState, TraineeFSAs} from './trainees/types'
import {initialTraineeState} from './trainees'
import {UiState, UiFSAs} from './ui/types'
import {TrainingFSAs} from './trainings/types'
import {initialUiState} from './ui/uiReducer'
import {createStore, applyMiddleware} from 'redux'
import {rootReducer} from './rootReducer'
import createSagaMiddleware from 'redux-saga'
import {rootSaga} from './sagas/rootSaga'

export interface ApplicationState {
  trainees: TraineeState
  ui: UiState
}

export const initialApplicationState: ApplicationState = {
  trainees: initialTraineeState,
  ui: initialUiState,
}

export type Action = TraineeFSAs | TrainingFSAs | UiFSAs

const sagaMiddleware = createSagaMiddleware()

export const store = createStore(
  rootReducer,
  initialApplicationState,
  applyMiddleware(sagaMiddleware)
)

sagaMiddleware.run(rootSaga)
