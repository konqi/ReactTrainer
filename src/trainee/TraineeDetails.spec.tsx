import {fireEvent, render} from '@testing-library/react'
import * as React from 'react'
import {TraineeDetails} from './TraineeDetails'
import {Provider} from 'react-redux'
import {
  createMockStore,
  TraineeBuilder,
  SessionBuilder,
} from '../__mocks__/store'
import {initialApplicationState, ApplicationState} from '../state'

describe('snapshot tests', () => {
  test('with no other session', () => {
    const trainee = new TraineeBuilder().build()
    const store = createMockStore({
      ...initialApplicationState,
      trainees: {
        [trainee.id]: trainee,
      },
    })

    const {unmount, baseElement} = render(
      <Provider store={store}>
        <TraineeDetails traineeId={'traineeId'} />
      </Provider>
    )

    expect(baseElement).toMatchSnapshot()

    unmount()
  })

  test('with session in past', () => {
    const trainee = new TraineeBuilder().build()
    const session = new SessionBuilder(trainee.id).build()
    trainee.sessionsRef = [session.id]
    const store = createMockStore({
      ...initialApplicationState,
      trainees: {
        [trainee.id]: trainee,
      },
      sessions: {[session.id]: session},
    })

    const {unmount, baseElement} = render(
      <Provider store={store}>
        <TraineeDetails traineeId={'traineeId'} />
      </Provider>
    )

    expect(baseElement).toMatchSnapshot()

    unmount()
  })

  test('with session in future', () => {})
  test('with sessions in past and future', () => {})
  test('with empty sessoins array')
})
