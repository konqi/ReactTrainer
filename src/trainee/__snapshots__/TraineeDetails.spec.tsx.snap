// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot tests with no other session 1`] = `
<body>
  <div>
    <div
      class="MuiContainer-root makeStyles-root-1 MuiContainer-maxWidthLg"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded"
      >
        <div
          class="MuiContainer-root MuiContainer-maxWidthLg"
        >
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3"
          >
            <div
              class="MuiGrid-root makeStyles-calendar-2 MuiGrid-item MuiGrid-grid-xs-4"
            />
            <div
              class="MuiGrid-root makeStyles-current-3 MuiGrid-item MuiGrid-grid-xs-4"
            >
              <div
                class="MuiPaper-root MuiPaper-elevation1 makeStyles-root-142 MuiPaper-rounded"
              >
                <div
                  class="MuiGrid-root makeStyles-calendar-143 MuiGrid-container MuiGrid-direction-xs-column"
                >
                  <div
                    class="MuiGrid-root makeStyles-center-144 MuiGrid-item"
                  >
                    <span
                      class="MuiTypography-root MuiTypography-h4"
                    >
                      16
                    </span>
                  </div>
                  <div
                    class="MuiGrid-root MuiGrid-item"
                  >
                    Juli
                  </div>
                </div>
              </div>
            </div>
            <div
              class="MuiGrid-root makeStyles-calendar-2 MuiGrid-item MuiGrid-grid-xs-4"
            />
          </div>
          <div
            class="MuiGrid-root MuiGrid-container"
          >
            <div
              class="MuiGrid-root makeStyles-form-4 MuiGrid-item MuiGrid-grid-xs-12"
            >
              <form>
                <div
                  class="MuiGrid-root makeStyles-center-175 MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-direction-xs-column"
                >
                  <div
                    class="MuiGrid-root MuiGrid-item"
                  >
                    <div
                      class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
                    >
                      <label
                        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled"
                        data-shrink="true"
                        for="TrainingDateInput"
                      >
                        Datum
                      </label>
                      <div
                        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
                      >
                        <fieldset
                          aria-hidden="true"
                          class="PrivateNotchedOutline-root-232 MuiOutlinedInput-notchedOutline"
                          style="padding-left: 8px;"
                        >
                          <legend
                            class="PrivateNotchedOutline-legend-233"
                            style="width: 0px;"
                          >
                            <span>
                              ​
                            </span>
                          </legend>
                        </fieldset>
                        <input
                          aria-invalid="false"
                          class="MuiInputBase-input MuiOutlinedInput-input"
                          id="TrainingDateInput"
                          type="date"
                          value="2019-07-16"
                        />
                      </div>
                    </div>
                  </div>
                  <div
                    class="MuiGrid-root MuiGrid-item"
                  >
                    <div
                      class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
                    >
                      <label
                        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled"
                        data-shrink="true"
                        for="TrainingPayedAmountInput"
                      >
                        Bezahlt
                      </label>
                      <div
                        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
                      >
                        <fieldset
                          aria-hidden="true"
                          class="PrivateNotchedOutline-root-232 MuiOutlinedInput-notchedOutline"
                          style="padding-left: 8px;"
                        >
                          <legend
                            class="PrivateNotchedOutline-legend-233"
                            style="width: 0px;"
                          >
                            <span>
                              ​
                            </span>
                          </legend>
                        </fieldset>
                        <input
                          aria-invalid="false"
                          class="MuiInputBase-input MuiOutlinedInput-input"
                          id="TrainingPayedAmountInput"
                          type="number"
                          value="15"
                        />
                      </div>
                    </div>
                  </div>
                  <div
                    class="MuiGrid-root MuiGrid-item"
                  >
                    <div
                      class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
                    >
                      <label
                        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
                        data-shrink="false"
                        for="TrainingDescriptionInput"
                      >
                        Beschreibung
                      </label>
                      <div
                        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-multiline MuiOutlinedInput-multiline"
                      >
                        <fieldset
                          aria-hidden="true"
                          class="PrivateNotchedOutline-root-232 MuiOutlinedInput-notchedOutline"
                          style="padding-left: 8px;"
                        >
                          <legend
                            class="PrivateNotchedOutline-legend-233"
                            style="width: 0.01px;"
                          >
                            <span>
                              ​
                            </span>
                          </legend>
                        </fieldset>
                        <textarea
                          aria-invalid="false"
                          class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMultiline MuiOutlinedInput-inputMultiline"
                          id="TrainingDescriptionInput"
                          rows="1"
                        />
                        <textarea
                          aria-hidden="true"
                          class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMultiline MuiOutlinedInput-inputMultiline"
                          readonly=""
                          style="visibility: hidden; position: absolute; overflow: hidden; height: 0px; width: 100%;"
                          tabindex="-1"
                        />
                      </div>
                    </div>
                  </div>
                  <div
                    class="MuiGrid-root MuiGrid-item"
                  >
                    <button
                      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-fullWidth"
                      tabindex="0"
                      type="submit"
                    >
                      <span
                        class="MuiButton-label"
                      >
                        Hinzufügen
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root"
                          focusable="false"
                          role="presentation"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
                          />
                        </svg>
                      </span>
                      <span
                        class="MuiTouchRipple-root"
                      />
                    </button>
                  </div>
                </div>
              </form>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`snapshot tests with session in past 1`] = `
<body>
  <div />
  <div>
    <div
      class="MuiContainer-root makeStyles-root-273 MuiContainer-maxWidthLg"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded"
      >
        <div
          class="MuiContainer-root MuiContainer-maxWidthLg"
        >
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3"
          >
            <div
              class="MuiGrid-root makeStyles-calendar-274 MuiGrid-item MuiGrid-grid-xs-4"
            >
              <div
                class="MuiPaper-root MuiPaper-elevation1 makeStyles-root-414 MuiPaper-rounded"
              >
                <div
                  class="MuiGrid-root makeStyles-calendar-415 MuiGrid-container MuiGrid-direction-xs-column"
                >
                  <div
                    class="MuiGrid-root makeStyles-center-416 MuiGrid-item"
                  >
                    <span
                      class="MuiTypography-root MuiTypography-h4"
                    >
                      16
                    </span>
                  </div>
                  <div
                    class="MuiGrid-root MuiGrid-item"
                  >
                    Juli
                  </div>
                </div>
              </div>
            </div>
            <div
              class="MuiGrid-root makeStyles-current-275 MuiGrid-item MuiGrid-grid-xs-4"
            >
              <div
                class="MuiPaper-root MuiPaper-elevation1 makeStyles-root-414 MuiPaper-rounded"
              >
                <div
                  class="MuiGrid-root makeStyles-calendar-415 MuiGrid-container MuiGrid-direction-xs-column"
                >
                  <div
                    class="MuiGrid-root makeStyles-center-416 MuiGrid-item"
                  >
                    <span
                      class="MuiTypography-root MuiTypography-h4"
                    >
                      16
                    </span>
                  </div>
                  <div
                    class="MuiGrid-root MuiGrid-item"
                  >
                    Juli
                  </div>
                </div>
              </div>
            </div>
            <div
              class="MuiGrid-root makeStyles-calendar-274 MuiGrid-item MuiGrid-grid-xs-4"
            />
          </div>
          <div
            class="MuiGrid-root MuiGrid-container"
          >
            <div
              class="MuiGrid-root makeStyles-form-276 MuiGrid-item MuiGrid-grid-xs-12"
            >
              <form>
                <div
                  class="MuiGrid-root makeStyles-center-447 MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-direction-xs-column"
                >
                  <div
                    class="MuiGrid-root MuiGrid-item"
                  >
                    <div
                      class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
                    >
                      <label
                        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled"
                        data-shrink="true"
                        for="TrainingDateInput"
                      >
                        Datum
                      </label>
                      <div
                        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
                      >
                        <fieldset
                          aria-hidden="true"
                          class="PrivateNotchedOutline-root-504 MuiOutlinedInput-notchedOutline"
                          style="padding-left: 8px;"
                        >
                          <legend
                            class="PrivateNotchedOutline-legend-505"
                            style="width: 0px;"
                          >
                            <span>
                              ​
                            </span>
                          </legend>
                        </fieldset>
                        <input
                          aria-invalid="false"
                          class="MuiInputBase-input MuiOutlinedInput-input"
                          id="TrainingDateInput"
                          type="date"
                          value="2019-07-16"
                        />
                      </div>
                    </div>
                  </div>
                  <div
                    class="MuiGrid-root MuiGrid-item"
                  >
                    <div
                      class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
                    >
                      <label
                        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled"
                        data-shrink="true"
                        for="TrainingPayedAmountInput"
                      >
                        Bezahlt
                      </label>
                      <div
                        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
                      >
                        <fieldset
                          aria-hidden="true"
                          class="PrivateNotchedOutline-root-504 MuiOutlinedInput-notchedOutline"
                          style="padding-left: 8px;"
                        >
                          <legend
                            class="PrivateNotchedOutline-legend-505"
                            style="width: 0px;"
                          >
                            <span>
                              ​
                            </span>
                          </legend>
                        </fieldset>
                        <input
                          aria-invalid="false"
                          class="MuiInputBase-input MuiOutlinedInput-input"
                          id="TrainingPayedAmountInput"
                          type="number"
                          value="15"
                        />
                      </div>
                    </div>
                  </div>
                  <div
                    class="MuiGrid-root MuiGrid-item"
                  >
                    <div
                      class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
                    >
                      <label
                        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
                        data-shrink="false"
                        for="TrainingDescriptionInput"
                      >
                        Beschreibung
                      </label>
                      <div
                        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-multiline MuiOutlinedInput-multiline"
                      >
                        <fieldset
                          aria-hidden="true"
                          class="PrivateNotchedOutline-root-504 MuiOutlinedInput-notchedOutline"
                          style="padding-left: 8px;"
                        >
                          <legend
                            class="PrivateNotchedOutline-legend-505"
                            style="width: 0.01px;"
                          >
                            <span>
                              ​
                            </span>
                          </legend>
                        </fieldset>
                        <textarea
                          aria-invalid="false"
                          class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMultiline MuiOutlinedInput-inputMultiline"
                          id="TrainingDescriptionInput"
                          rows="1"
                        />
                        <textarea
                          aria-hidden="true"
                          class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMultiline MuiOutlinedInput-inputMultiline"
                          readonly=""
                          style="visibility: hidden; position: absolute; overflow: hidden; height: 0px; width: 100%;"
                          tabindex="-1"
                        />
                      </div>
                    </div>
                  </div>
                  <div
                    class="MuiGrid-root MuiGrid-item"
                  >
                    <button
                      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-fullWidth"
                      tabindex="0"
                      type="submit"
                    >
                      <span
                        class="MuiButton-label"
                      >
                        Hinzufügen
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root"
                          focusable="false"
                          role="presentation"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
                          />
                        </svg>
                      </span>
                      <span
                        class="MuiTouchRipple-root"
                      />
                    </button>
                  </div>
                </div>
              </form>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
